domain/blockdevice/watcher_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package blockdevice_test
```

Comment:
This needs to go somewhere else. It's not a testing concern of this package. We are either testing the service or the state and should be writing tests in those packages to that effect.

Reply:
As discussed live, this is an integration test which we'll keep here